test/api/jd_by_pulse_test.go

```diff
@@ -0,0 +1,228 @@
+		fromJdIdx := 1
+		queryParams := client.JetDropsByPulseNumberOpts{
+			Offset:        optional.NewInt32(int32(10)),
+			Limit:         optional.NewInt32(int32(10)),
+			FromJetDropId: optional.NewString(jdList.Result[fromJdIdx].JetDropId),
+		}
+		response, err := c.JetDropsByPulseNumber(t, int64(pn), &queryParams)
+		require.NoError(t, err)
+		require.Len(t, response.Result, 10)
+		require.Equal(t, int64(jetDropsCount-fromJdIdx), response.Total)
+	})
+	t.Run("offset 1", func(t *testing.T) {
+		t.Log("C5230 Get Jet drops by Pulse number with offset = 1")
+		offset := 1
+		queryParams := client.JetDropsByPulseNumberOpts{
+			Offset: optional.NewInt32(int32(offset)),
+		}
+		response, err := c.JetDropsByPulseNumber(t, int64(pn), &queryParams)
+		require.NoError(t, err)
+		require.Len(t, response.Result, defaultLimit)
+		require.Equal(t, int64(jetDropsCount), response.Total)
+		require.Equal(t, jdList.Result[1], response.Result[0])
+	})
+	t.Run("offset out of range", func(t *testing.T) {
+		t.Log("C5231 Get Jet drops by Pulse number with offset out of range")
+		offset := jetDropsCount
+		queryParams := client.JetDropsByPulseNumberOpts{
+			Offset: optional.NewInt32(int32(offset)),
+		}
+		response, err := c.JetDropsByPulseNumber(t, int64(pn), &queryParams)
+		require.NoError(t, err)
+		require.Len(t, response.Result, 0)
+	})
+	t.Run("with FromJetDropId and Offset", func(t *testing.T) {
+		t.Log("C5232 Get Jet drops by Pulse number with FromJetDropId and Offset")
+		fromJdIdx, offset := 10, 10
+		queryParams := client.JetDropsByPulseNumberOpts{
+			FromJetDropId: optional.NewString(jdList.Result[fromJdIdx].JetDropId),
+			Offset:        optional.NewInt32(int32(offset)),
+		}
+		response, err := c.JetDropsByPulseNumber(t, int64(pn), &queryParams)
+		require.NoError(t, err)
+		require.Equal(t, int64(jetDropsCount-fromJdIdx), response.Total)
+		require.Equal(t, jdList.Result[fromJdIdx+offset], response.Result[0])
+	})
+	t.Run("FromJetDropId invalid", func(t *testing.T) {
+		t.Log("C5233 Get Jet drops by Pulse number with invalid FromJetDropId")
+		queryParams := client.JetDropsByPulseNumberOpts{
+			FromJetDropId: optional.NewString("%^&Qwerty!@#$%123"),
+		}
+		_, err := c.JetDropsByPulseNumber(t, int64(pn), &queryParams)
+		require.Error(t, err)
+		require.Equal(t, "400 Bad Request", err.Error())
+	})
+	t.Run("FromJetDropId empty", func(t *testing.T) {
+		t.Log("C5234 Get Jet drops by Pulse number with empty FromJetDropId")
+		queryParams := client.JetDropsByPulseNumberOpts{
+			FromJetDropId: optional.NewString(""),
+		}
+		_, err := c.JetDropsByPulseNumber(t, int64(pn), &queryParams)
+		require.Error(t, err)
+		require.Equal(t, "400 Bad Request", err.Error())
+	})
+	t.Run("FromJetDropId too big", func(t *testing.T) {
+		t.Log("C5235 Get Jet drops by Pulse number with too big FromJetDropId")
+
+		s := strconv.FormatInt(testutils.RandNumberOverRange(math.MaxInt32, math.MaxInt32+1), 10)
+		for i := 0; i < 10; i++ {
+			s += s
+		}
```

Comment:
if it's an integer you can just use multiplication `s = s*10; `, isn't?

Reply:
it's a string and the goal is to get very big string with ~1000 chars